src/tasks/python/PythonTaskHelper.ts

```diff
@@ -0,0 +1,193 @@
+    private static readonly defaultLabels: { [key: string]: string } = {
+        "com.microsoft.created-by": "visual-studio-code"
+    };
+
+    public async getDockerBuildOptions(context: DockerBuildTaskContext, buildDefinition: DockerBuildTaskDefinition): Promise<DockerBuildOptions> {
+        const buildOptions = buildDefinition.dockerBuild;
+
+        /* eslint-disable no-template-curly-in-string */
+        buildOptions.context = buildOptions.context || "${workspaceFolder}";
+        buildOptions.dockerfile = buildOptions.dockerfile || "${workspaceFolder}/Dockerfile";
+        buildOptions.tag = buildOptions.tag || getDefaultImageName(context.folder.name);
+        buildOptions.labels = buildOptions.labels || PythonTaskHelper.defaultLabels;
+
+        return buildOptions;
+    }
+
+    public async getDockerRunOptions(context: DockerRunTaskContext, runDefinition: DockerRunTaskDefinition): Promise<DockerRunOptions> {
+        // tslint:disable no-unsafe-any
+        const helperOptions : PythonTaskRunOptions = runDefinition.python || {};
+        const runOptions : DockerRunOptions = runDefinition.dockerRun;
+
+        const target: PythonFileTarget | PythonModuleTarget = helperOptions.file
+            ? { file: helperOptions.file, }
+            : { module: helperOptions.module };
+
+        const launcherCommand : string = PythonExtensionHelper.getRemotePtvsdCommand(
+            target,
+            helperOptions.args,
+            {
+                host: "0.0.0.0",
+                port: helperOptions.debugPort || PythonDefaultDebugPort,
+                wait: helperOptions.wait === undefined ? true : helperOptions.wait
+            }
+        );
+
+        const launcherFolder: string = PythonExtensionHelper.getLauncherFolderPath();
+
+        runOptions.image = inferImageName(
+            runDefinition,
+            context,
+            context.folder.name
+        );
+
+        runOptions.containerName = runOptions.containerName || getDefaultContainerName(context.folder.name);
+
+        const dbgLogsFolder = path.join(os.tmpdir(), context.folder.name);
+
+        // The debugger will complain if the logs directory does not exist.
+        if (!fse.existsSync(dbgLogsFolder)) {
+            fse.emptyDirSync(dbgLogsFolder);
+        }
+
+        // User input is honored in all of the below.
+        runOptions.volumes = this.inferVolumes(runOptions, launcherFolder, dbgLogsFolder);
+        runOptions.ports = this.inferPorts(runOptions, helperOptions);
+        runOptions.entrypoint = runOptions.entrypoint || "python";
+        runOptions.command = runOptions.command || launcherCommand;
+
+        runOptions.env = this.addDebuggerEnvironmentVars(runOptions.env);
+        runOptions.portsPublishAll = runOptions.portsPublishAll || true;
+
+        return runOptions;
+    }
+
+    public async provideDockerBuildTasks(context: DockerTaskScaffoldContext): Promise<DockerBuildTaskDefinition[]> {
```

Comment:
Nit: the other platforms have `provideDocker*Tasks` above the `getDocker*Options`. Conceptually, you scaffold before you run.